libcalico-go/lib/names/workloadendpointstatus.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2019 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package names
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/projectcalico/calico/felix/proto"
+
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	fieldOrchestratorID = iota
+	fieldWorkloadID
+	fieldEndpointID
+)
+const (
+	separator = "-"
```

Comment:
I think `-` is a valid char in the various parts of the ID so it's no good as a separator.  You could use `/` and URL-encode it.  We know `/` is OK because we use it in our etcd names.  Otherwise you could inspect the validation regexes to find a good separator that's not already in use.

Reply:
Now using spaces as separators which the FS is happy about. Im also percent-encoding/decoding slashes using PathEncode std function